components/automate-ui/src/app/modules/infra-proxy/create-environment-modal/create-environment-modal.component.spec.ts

```diff
@@ -0,0 +1,179 @@
+import { ReactiveFormsModule, FormBuilder } from '@angular/forms';
+import { StoreModule, Store } from '@ngrx/store';
+import { NgrxStateAtom, ngrxReducers, runtimeChecks } from 'app/ngrx.reducers';
+import { ComponentFixture, TestBed, waitForAsync
+  } from '@angular/core/testing';
+import { MockComponent } from 'ng2-mock-component';
+
+import { CreateEnvironmentModalComponent } from './create-environment-modal.component';
+import { TelemetryService } from 'app/services/telemetry/telemetry.service';
+import { CreateEnvironmentSuccess, CreateEnvironmentFailure } from 'app/entities/environments/environment.action';
+import { HttpErrorResponse } from '@angular/common/http';
+import { HttpStatus } from 'app/types/types';
+import { EventEmitter } from '@angular/core';
+import { HttpClient, HttpHandler } from '@angular/common/http';
+
+
+class MockTelemetryService {
+  track() { }
+}
+
+describe('CreateEnvironmentModalComponent', () => {
+  let component: CreateEnvironmentModalComponent;
+  let fixture: ComponentFixture<CreateEnvironmentModalComponent>;
+
+  beforeEach( waitForAsync(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        MockComponent({ selector: 'chef-button', inputs: ['disabled'] }),
+        MockComponent({ selector: 'chef-loading-spinner' }),
+        MockComponent({ selector: 'chef-form-field' }),
+        MockComponent({ selector: 'chef-error' }),
+        MockComponent({ selector: 'chef-toolbar' }),
+        MockComponent({ selector: 'app-infra-tab-change' }),
+        MockComponent({ selector: 'app-infra-tab', inputs: ['active', 'disabled'] }),
+
+        MockComponent({ selector: 'chef-modal',
+          inputs: ['visible']
+        }),
+        CreateEnvironmentModalComponent
+      ],
+      providers: [
+        { provide: TelemetryService, useClass: MockTelemetryService },
+        HttpClient, HttpHandler
+      ],
+      imports: [
+        ReactiveFormsModule,
+        StoreModule.forRoot(ngrxReducers, { runtimeChecks })
+      ]
+    })
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(CreateEnvironmentModalComponent);
+    component = fixture.componentInstance;
+    component.detailsFormGroup = new FormBuilder().group({
+      name: ['', null],
+      description: ['', null]
+    });
+    component.defaultAttrFormGroup = new FormBuilder().group({
+      default: ['', null]
+    });
+    component.overrideAttrFormGroup = new FormBuilder().group({
+      override: ['', null]
+    });
+    component.openEvent = new EventEmitter();
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+
+  describe('create environment', () => {
+    let store: Store<NgrxStateAtom>;
+    const environment = {
```

Comment:
Can we strongly type this constant? That provides a guarantee that all the properties are valid.

Reply:
Not sure why you marked this resolved; it still has no type.

Reply:
yes, i have updated the changes